File: packages/python/src/trend_buddy_shared/contracts/domain/runtime.py
from dataclasses import dataclass
from datetime import datetime
from enum import StrEnum


class LiveRunStatus(StrEnum):
    PENDING = "pending"
    RUNNING = "running"
    PAUSED = "paused"
    STOPPED = "stopped"
    FAILED = "failed"


@dataclass
class LiveRun:
    id: str
    strategy_id: str
    instrument_id: str
    status: LiveRunStatus
    started_at: datetime
    updated_at: datetime

File: packages/go/contracts/domain/runtime.go
package domain

import "time"

type LiveRunID string

type LiveRunStatus string

const (
	LiveRunStatusPending LiveRunStatus = "pending"
	LiveRunStatusRunning LiveRunStatus = "running"
	LiveRunStatusPaused  LiveRunStatus = "paused"
	LiveRunStatusStopped LiveRunStatus = "stopped"
	LiveRunStatusFailed  LiveRunStatus = "failed"
)

type LiveRun struct {
	ID           LiveRunID     `json:"id"`
	StrategyID   StrategyID    `json:"strategy_id"`
	InstrumentID InstrumentID  `json:"instrument_id"`
	Status       LiveRunStatus `json:"status"`
	StartedAt    time.Time     `json:"started_at"`
	UpdatedAt    time.Time     `json:"updated_at"`
}
